lib/puppet/util/execution.rb

```diff
@@ -188,7 +188,8 @@ def self.execute(command, options = NoOptionsSpecified)
 
     begin
       stdin = Puppet::FileSystem.open(options[:stdinfile] || null_file, nil, 'r')
-      stdout = options[:squelch] ? Puppet::FileSystem.open(null_file, nil, 'w') : Puppet::FileSystem::Uniquefile.new('puppet')
+      reader, writer = IO.pipe
```

Comment:
64KB test fails, need to read output in a loop to clear the buffer before waiting on the process.